iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/agent/task/meta/PipeMeta.java

```diff
@@ -61,6 +65,49 @@ public PipeTemporaryMeta getTemporaryMeta() {
     return temporaryMeta;
   }
 
+  public boolean matchSqlDialect(final boolean isTableModel) {
+    return matchSqlDialect(SystemConstant.fetchSqlDialectValue(isTableModel));
+  }
+
+  public boolean matchSqlDialect(final String sqlDialect) {
+    if (Objects.isNull(sqlDialect)) {
+      return true;
+    }
+    return matchBoth()
+        || sqlDialect.equalsIgnoreCase(
+            getStaticMeta()
+                .getExtractorParameters()
+                .getStringOrDefault(
+                    SystemConstant.SQL_DIALECT_KEY, SystemConstant.SQL_DIALECT_TREE_VALUE));
+  }
+
+  private boolean matchBoth() {
+    return // 1. 'mode.double-living' is set to true
+    getStaticMeta()
+            .getExtractorParameters()
+            .getBooleanOrDefault(
+                Arrays.asList(
+                    PipeExtractorConstant.EXTRACTOR_MODE_DOUBLE_LIVING_KEY,
+                    PipeExtractorConstant.SOURCE_MODE_DOUBLE_LIVING_KEY),
+                PipeExtractorConstant.EXTRACTOR_MODE_DOUBLE_LIVING_DEFAULT_VALUE)
+        ||
+        // 2. 'capture.tree' and 'capture.table' is set to true
+        (getStaticMeta()
+                .getExtractorParameters()
+                .getBooleanOrDefault(
+                    Arrays.asList(
+                        PipeExtractorConstant.EXTRACTOR_CAPTURE_TREE_KEY,
```

Comment:
What if capture.tree == true and sql-dialect == table? Now a pipe will be shown in the tree model if "capture.tree" and "capture.table" is true, but sql-dialect is "table"; will not in this case if "capture.table" is false...